project-base/storefront/components/Blocks/SortingBar/SortingBar.tsx

```diff
@@ -105,49 +109,65 @@ export const SortingBar: FC<SortingBarProps> = ({ sorting, totalCount, customSor
                 ) : (
                     <>
                         <div className="flex vl:gap-3">
-                            {sortOptions.map((sortOption, index) => (
-                                <SortingBarItem
-                                    key={sortOption}
-                                    onClick={() => updateSort(sortOption)}
-                                    dataTestId={TEST_IDENTIFIER + '-' + index}
-                                >
-                                    <SortingBarItemLink isActive={sortOption === selectedSortOption}>
-                                        <span>{sortOptionsLabels[sortOption]}</span>
-                                    </SortingBarItemLink>
-                                </SortingBarItem>
-                            ))}
+                            {sortOptions.map((sortOption, index) => {
+                                // eslint-disable-next-line @typescript-eslint/no-unused-vars
+                                const { page, ...queriesWithoutPage } = getFilteredQueries(router.query);
+                                const sortParams = new URLSearchParams({
+                                    ...queriesWithoutPage,
+                                    sort: sortOption,
+                                }).toString();
+                                const sortHref = `${asPathWithoutQueryParams}?${sortParams}`;
+
+                                return (
+                                    <SortingBarItem
+                                        key={sortOption}
+                                        isActive={sortOption === selectedSortOption}
+                                        onClick={() => updateSort(sortOption)}
+                                        dataTestId={TEST_IDENTIFIER + '-' + index}
+                                        href={sortHref}
```

Comment:
It is a little bit weird. We are setting the href, then preventing its behavior, and handling it manually. I probably understand the semantic reasoning behind that, but I am not sure if thats the best way to handle it. Cannot the buttons just be links? Or if we want to use the `updateSort` function (preferred by me), cannot it just be buttons?

Reply:
Alright then